dracut/modules.d/99kiwi-lib/kiwi-partitions-lib.sh

```diff
@@ -307,14 +307,19 @@ function relocate_gpt_at_end_of_disk {
     fi
 }
 
-function disk_has_unallocated_space {
+function disk_is_fully_allocated {
     local disk_device=$1
     local pt_table_type
     pt_table_type=$(get_partition_table_type "${disk_device}")
     if [ "${pt_table_type}" = "dos" ];then
-        sfdisk --verify "${disk_device}" 2>&1 | grep -q "unallocated"
+        sfdisk --verify "${disk_device}" 2>&1 |\
+            grep -q "unallocated" && false
```

Comment:
This is always false:

```
seife@strolchi:~> echo "hello" | grep -q hello && false
seife@strolchi:~> echo $?
1
seife@strolchi:~> echo "hello" | grep -q hrmpf && false
seife@strolchi:~> echo $?
1
```
This is probably best written out explicitly:
```
  if sfdisk --verify $disk_device 2>&1 | grep -q "unallocated"; then
    false
  else
    true
  fi
```
Yes, it's long, but it is easily readable and works ;-)

There is also this (I found it on Stackoverflow):
```
seife@strolchi:~> ! echo "hello" | grep -q hello
seife@strolchi:~> echo $?
1
seife@strolchi:~> ! echo "hello" | grep -q hmpf
seife@strolchi:~> echo $?
0
```
But I personally think it's slightly obscure... probably many people will stumble over it and wonder what it does.

Reply:
I'm so stupid, thanks for pointing it out. Fortunately bash has !(not), that should make it relatively readable